File: backend/src/models/user.model.ts
export type UserRole = 'customer' | 'restaurant' | 'rider' | 'admin';
export type UserStatus = 'active' | 'suspended';

export interface User {
  id: string;
  email: string;
  password_hash: string;
  role: UserRole;
  display_name: string | null;
  phone: string | null;
  profile_photo_url: string | null;
  status: UserStatus;
  created_at: Date;
  updated_at: Date;
}

export interface PublicUser {
  id: string;
  email: string;
  role: UserRole;
  display_name: string | null;
  phone: string | null;
  profile_photo_url: string | null;
  status: UserStatus;
  created_at: Date;
}

File: mobile/restaurant/lib/features/auth/models/user_model.dart
class UserModel {
  final String id;
  final String email;
  final String role;
  final String? displayName;
  final String? phone;
  final String? profilePhotoUrl;
  final String status;

  const UserModel({
    required this.id,
    required this.email,
    required this.role,
    this.displayName,
    this.phone,
    this.profilePhotoUrl,
    required this.status,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
    id: json['id'] as String,
    email: json['email'] as String,
    role: json['role'] as String,
    displayName: json['display_name'] as String?,
    phone: json['phone'] as String?,
    profilePhotoUrl: json['profile_photo_url'] as String?,
    status: json['status'] as String,
  );
}
